src/corelibs/U2Core/src/models/DocumentModel.cpp

```diff
@@ -313,6 +313,8 @@ void Document::_addObject(GObject* obj) {
 bool Document::removeObject(GObject* obj, DocumentObjectRemovalMode removalMode) {
     SAFE_POINT(df->isObjectOpSupported(this, DocumentFormat::DocObjectOp_Remove, obj->getGObjectType()), "Unsupported format operation", false);
 
+    emit si_removeRelations(obj);
```

Comment:
This is bad design: when object is removed the Document should emit si_objectRemoved.
If this phase has some stages: before/after: rename the signal to si_beforeObjectRemoved